pkg/setting/setting_unified_alerting.go

```diff
@@ -240,6 +242,7 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
 	uaCfg.HARedisUsername = ua.Key("ha_redis_username").MustString("")
 	uaCfg.HARedisPassword = ua.Key("ha_redis_password").MustString("")
 	uaCfg.HARedisDB = ua.Key("ha_redis_db").MustInt(0)
+	uaCfg.HARedisMaxConns = ua.Key("ha_redis_max_conns").MustInt(alertmanagerRedisDefaultMaxConns)
```

Comment:
should we have lower and upper bounds and allow user to only set it within the bounds?

Reply:
- It probably crashes if negative - I'll add some logic to coerce this to zero.
- If zero, it'll use the redis client's default (i.e. there is no way to make it infinite), I think this is desirable
- `[database] max_idle_conns` does not seem to have a forced upper bound either, only a default of 100 - this doesn't feel extremely necessary.

Reply:
Added, with a slight difference - negative is coerced to grafana's default, not go-redis's default.

Reply:
thanks. I think in this it makes sense to handle only negative